plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -107,11 +105,40 @@ export class TaskWorker {
     });
   }
 
+  private async registerAbortable(task: TaskManager) {
+    const taskId = await task.getWorkspaceName();
+    let shouldUnsubscribe = false;
+
+    const subscription = this.options.taskBroker
+      .event$({ taskId, after: undefined })
+      .subscribe({
+        error: _ => {
+          shouldUnsubscribe = true;
+        },
+        next: ({ events }) => {
+          for (const event of events) {
+            if (event.type === 'aborted') {
+              task.abortContext.abort();
+              shouldUnsubscribe = true;
+            }
+
+            if (event.type === 'completion') {
+              shouldUnsubscribe = true;
+            }
+          }
+          if (shouldUnsubscribe) {
+            subscription.unsubscribe();
+          }
+        },
+      });
+  }
+
   start() {
     (async () => {
       for (;;) {
         await this.onReadyToClaimTask();
         const task = await this.options.taskBroker.claim();
+        await this.registerAbortable(task as unknown as TaskManager);
```

Comment:
Not fond of the cast here - that might just shadow over actual problems down the line, especially if the manager is replaceable. Should be possible to find the right abort controller anyway.

Honestly, why do we even listen to a broker event stream? The broker is the one that's responsible for responding to the claim(). Why doesn't the broker return a context that has an `abortSignal: AbortSignal` in it? The broker is the one that knows when it's time to abort, isn't it, based on its internal functionality? So it can hold the corresponding `AbortController` internally, and trigger it at the right time which has the side effect that this worker also gets informed about it through the signal

Reply:
Ok, I will try to think about other way how to access abort controller if that's possible.

Regarding listening an event steam. It's related to a running Backstage in a cluster, where the task is running in instance A, but user can abort it from instance B.

Reply:
![abort-task drawio](https://user-images.githubusercontent.com/2265094/219684820-c6e4f602-9eac-499d-b4f3-6fc9dc3fbf6a.png)

Reply:
Ah, right. But no matter how the information travels, the task broker in Backend1 still is the one who would know how to abort the task it holds. So it should be kept as a broker-internal concern, and it only injecting a signal (not controller) into its in-memory task

Reply:
Yeah, I agree. I already have a working solution for that. 
Have to fix a test yet.